backend/service/authn/oidc.go

```diff
@@ -175,6 +184,15 @@ func (p *OIDCProvider) Verify(ctx context.Context, rawToken string) (*Claims, er
 		return nil, err
 	}
 
+	// If the token doesn't exist in the token storage anymore, it must have been revoked.
+	// Fail verification in this case.
+	if p.tokenStorage != nil {
+		token, err := p.tokenStorage.Read(ctx, claims.Subject, clutchProviderName)
```

Comment:
can we add a `TODO(perf/maybe):` somewhere to cache tokens in-memory for `min(60s, timeUntilexpiry)`? once we have service accounts we could have high programmatic call volume which would roundtrip the DB before each call. thoughts on this scenario?

Reply:
Yeah I can imagine this being necessary, and we'd probably also want to include tokens whose lookup failed (and indicate this) to prevent a client using a revoked token from having to hit the DB every time. Will add a TODO